ResearchKit/Common/ORKSurveyAnswerCellForLocation.m

```diff
@@ -0,0 +1,172 @@
+ OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#import "ORKSurveyAnswerCellForLocation.h"
+#import <MapKit/MapKit.h>
+#import <AddressBookUI/AddressBookUI.h>
+#import "ORKAnswerTextField.h"
+#import "ORKHelpers.h"
+#import "ORKAnswerFormat_Internal.h"
+#import "ORKQuestionStep_Internal.h"
+#import "ORKLocationSelectionView.h"
+
+@interface ORKSurveyAnswerCellForLocation () <ORKLocationSelectionViewDelegate>
+    
+@end
+
+@implementation ORKSurveyAnswerCellForLocation {
+    ORKLocationSelectionView *_selectionView;
+}
+
+- (BOOL)becomeFirstResponder {
+    return [_selectionView becomeFirstResponder];
+}
+
+- (void)setStep:(ORKQuestionStep *)step {
+    [super setStep:step];
+}
+
++ (CGFloat)suggestedCellHeightForView:(UIView *)view {
+    return 282;
+}
+
+- (void)prepareView {
+    _selectionView = [[ORKLocationSelectionView alloc] init];
+    _selectionView.delegate = self;
+    _selectionView.tintColor = self.tintColor;
+    [_selectionView showMapViewAnimated:NO];
+    [self addSubview:_selectionView];
+    
+    NSDictionary *views = NSDictionaryOfVariableBindings(_selectionView);
+    ORKEnableAutoLayoutForViews([views allValues]);
+    
+    [self addConstraints:[NSLayoutConstraint constraintsWithVisualFormat:@"V:|[_selectionView]|"
+                                                                 options:NSLayoutFormatDirectionLeadingToTrailing
+                                                                 metrics:nil
+                                                                   views:views]];
+    [self addConstraints:[NSLayoutConstraint constraintsWithVisualFormat:@"H:|[_selectionView]|"
+                                                                 options:NSLayoutFormatDirectionLeadingToTrailing
+                                                                 metrics:nil
+                                                                   views:views]];
+    
+    NSLayoutConstraint *resistsCompressingMapConstraint = [NSLayoutConstraint constraintWithItem:_selectionView
+                                                                                       attribute:NSLayoutAttributeWidth
+                                                                                       relatedBy:NSLayoutRelationGreaterThanOrEqual
+                                                                                          toItem:nil
+                                                                                       attribute:NSLayoutAttributeNotAnAttribute
+                                                                                      multiplier:1.0
+                                                                                        constant:20000.0];
+    resistsCompressingMapConstraint.priority = UILayoutPriorityDefaultHigh;
+    [self addConstraint:resistsCompressingMapConstraint];
```

Comment:
Move constraints into `setUpConstraints` and call `activeConstraints` instead of adding them individually.